src/main/java/ch/njol/skript/variables/FlatFileStorage.java

```diff
@@ -387,23 +436,23 @@ public final void saveVariables(final boolean finalSave) {
 						pw.println();
 						pw.flush();
 						pw.close();
-						FileUtils.move(tempFile, f, true);
-					} catch (final IOException e) {
-						Skript.error("Unable to make a final save of the database '" + databaseName + "' (no variables are lost): " + ExceptionUtils.toString(e)); // FIXME happens at random - check locks/threads
-					} finally {
-						if (pw != null)
-							pw.close();
+						FileUtils.move(tempFile, file, true);
+					} catch (IOException e) {
+						Skript.error("Unable to make a final save of the database '" + databaseName +
+								"' (no variables are lost): " + ExceptionUtils.toString(e));
+						// FIXME happens at random - check locks/threads
```

Comment:
I wonder if this is related to the aforementioned problem of obtaining one lock without the other? I will look into this later.